### PR TITLE
extend hooks

### PR DESCRIPTION
**Pull request by cn-tools**
I added two new hooks to have more options for influencing with extensions.

### CODE REVIEW

**Comment by aledeg** [comment]
Thank you for your contribution.
Could you explain what are the use case for those new hooks? I guess that if you add those, you want new features for your extensions.

**Comment by cn-tools** [comment]
Dear @aledeg !
I develop a new extension who should change the url in the add process. My idea is to change a YouTube Channel URL into an YouTube RSS Feed URL. (hook->'check_url_before_add')

The hook 'feed_before_actualize' is only a abstract idea :)

Have a look:
https://github.com/cn-tools/cntools_FreshRssExtensions

File: lib/Minz/ExtensionManager.php [hunk 1]
```diff
@@ -23,10 +23,18 @@ class Minz_ExtensionManager {
 			'list' => array(),
 			'signature' => 'OneToOne',
 		),
+		'check_url_before_add' => array(  // function($url) -> Url | null
+			'list' => array(),
+			'signature' => 'OneToOne',
+		),
```
Comment (aledeg) [inline lib/Minz/ExtensionManager.php]: Move that definition above to keep the list ordered.

File: lib/Minz/ExtensionManager.php [hunk 2]
```diff
@@ -23,10 +23,18 @@ class Minz_ExtensionManager {
...
 		'feed_before_insert' => array(  // function($feed) -> Feed | null
 			'list' => array(),
 			'signature' => 'OneToOne',
 		),
+		'feed_before_actualize' => array(  // function($feed) -> Feed | null
+			'list' => array(),
+			'signature' => 'OneToOne',
+		),
```
Comment (aledeg) [inline lib/Minz/ExtensionManager.php]: Same here

File: app/Controllers/feedController.php [hunk 3]
```diff
@@ -274,6 +280,12 @@ public static function actualizeFeed($feed_id, $feed_url, $force, $simplePiePush
 		$updated_feeds = 0;
 		$nb_new_articles = 0;
 		foreach ($feeds as $feed) {
+			/** @var FreshRSS_Feed $feed */
+			$feed = Minz_ExtensionManager::callHook('feed_before_actualize', $feed);
+			if ($feed === null) {
```
Comment (aledeg) [inline app/Controllers/feedController.php]: Use `null === $feed` instead

File: app/Controllers/feedController.php [hunk 4]
```diff
@@ -47,6 +47,12 @@ public static function addFeed($url, $title = '', $cat_id = 0, $new_cat_name = '
 
 		$url = trim($url);
 
+		/** @var $url */
+		$url = Minz_ExtensionManager::callHook('check_url_before_add', $url);
+		if ($url === null) {
```
Comment (aledeg) [inline app/Controllers/feedController.php]: Use `null === $feed` instead

**Review by aledeg — CHANGES_REQUESTED** [review]
Apart from minor changes, this looks good to me.

**Comment by cn-tools** [comment]
thx 4 comments
change typography ... done

**Review by aledeg — APPROVED** [review]

**Comment by Alkarex** [comment]
@Frenzie I am not up to date on how to change the documentation now; we should add the new hooks to the documentation in https://github.com/FreshRSS/FreshRSS/blob/master/docs/en/developers/03_Backend/05_Extensions.md#the-hooks-system

**Comment by Frenzie** [comment]
@Alkarex The English docs are the source, then there's a little helper script to generate the rest. Unfortunately you need po4a 0.56 which may not be readily available yet. That's just these two? I can submit a PR with that.

**Comment by Alkarex** [comment]
Yes please @Frenzie 👍

**Pull request merged**